server/views/cms/documents.py

```diff
@@ -29,42 +30,53 @@
 
 BASE_URL = BASE_URL.rstrip('/')
 
+
+def fetch_cms_content(url, application_name=None, params=None):
+    """Helper function to make API requests with consistent headers"""
+    try:
+        headers = {
+            'Authorization': f'users API-Key {API_KEY}',
+            'Content-Type': 'application/json'
+        }
+        
+        if application_name:
+            headers['Cs-App'] = application_name
+    
+        escaped_args = {k: html.escape(v) for k, v in (params or {}).items()}
+
+        response = requests.get(url, params=escaped_args, headers=headers)
+        response.raise_for_status()
+        return response.json(), response.status_code
+
+    except requests.RequestException as e:
+        logger.error(f'Request failed: {str(e)}')
+        return jsonify({'message': 'An internal error has occurred.'}), 500
+    
+    except ValueError as e:
+        logger.error(f'Invalid JSON response: {str(e)}')
+        return jsonify({'message': 'An internal error has occurred.'}), 500
+    
+    except Exception as e:
+        logger.error(f'Unexpected error: {str(e)}')
+        return jsonify({'message': 'An internal error has occurred.'}), 500
```

Comment:
```suggestion
    headers = {
        'Authorization': f'users API-Key {API_KEY}',
        'Content-Type': 'application/json'
    }
    if application_name:
       headers['Cs-App'] = application_name
    try:
        escaped_args = {k: html.escape(v) for k, v in (params or {}).items()}
        response = requests.get(url, params=escaped_args, headers=headers)
        return response.json(), response.status_code
    except requests.RequestException:
        logger.exception('Request failed')
    except ValueError:
        logger.exception('Invalid JSON response')    
    except Exception:
        logger.exception('Unexpected error')
    return jsonify({'message': 'An internal error has occurred.'}), 500
```

1. In a `try`/`except`, `logger.exception` includes `e` by default (unlike `logger.error`).
2. Do we need `response.raise_for_status()`?
3. Reduce empty lines

Reply:
Cool, made the changes

Regarding 2. `response.raise_for_status()`:
I think we we do need it, the motive was to throw a `RequestException` for the 404 response that is coming with HTML content (from `charter.africa`)...but I'm also open to let it throw an Invalid JSON response

Reply:
>I think we we do need it, the motive was to throw a RequestException for 
>

Cool, cool... we can leave it there for better logs I guess.